packages/@aws-cdk/aws-codebuild/test/test.project.ts

```diff
@@ -1231,6 +1231,110 @@ export = {
         test.done();
       },
 
+      "when provided as a Token referencing a different account, adds permission to decrypt keys in the Secret's account"(test: Test) {
+        // GIVEN
+        const stack = new cdk.Stack();
+        const secretToken = cdk.Lazy.string({ produce: () => 'arn:aws:secretsmanager:us-west-2:901234567890:secret:mysecret' });
```

Comment:
In my use case, we get a SecretArn as a token and pass it into the environmentVariables. The SecretArn is from a secret living in a different account.

The 'work around' we currently do is the following:
``` ts
const parsedArn = Arn.parse(token);
const kmsKey = fromKeyArn(this, 'KmsKey', cdk.Arn.format({
      account: parsedArn.account,
      region: parsedArn.region,
      service: 'kms',
      resource: 'key',
      sep: '/',
      resourceName: '*',
    }, this);
kmsKey.grantDecrypt(project);
```

Reply:
> In my use case, we get a SecretArn as a token and pass it into the environmentVariables.

Can you specify why is the ARN passed as a Token?

Reply:
So what we do is we get the secretArn from another resource that we create in the stack, which creates the secret in a different account.
If I am not mistaken this should be the reason why it is passed as a token.

Reply:
So you pass a `Secret` instance between Stacks, and the Stacks are in different accounts, and then you use `secret.secretArn` in the CodeBuild Project's environment variables?

Reply:
Yeah I think you captured it pretty good. It's realized through a custom resource.

Reply:
I thought I understood what you meant, but then you lost me again 🙂.

So you're _not_ passing instances of `Secret` between Stacks? If not, can you show the relevant parts of your code, and how they work together?

Reply:
No, no you got it exactly right. We are passing instances of the secret between stack 👍 sorry for the confusion!

Reply:
Ok, that's good 🙂.

So, this is interesting. In theory, the `Secret` _should_ contain the account it belongs to. So, if we allowed passing it as the value of the environment variable, like this:

```ts
        new codebuild.PipelineProject(stack, 'Project', {
          environmentVariables: {
            'ENV_VAR1': {
              type: codebuild.BuildEnvironmentVariableType.SECRETS_MANAGER,
              value: mySecret,
            },
          },
```

We _could_ gate adding the cross-account permissions on that...

Thoughts on this @Kruspe?

Reply:
> We could gate adding the cross-account KMS permissions on that...

I am not sure what you mean by gate adding. Can you explain?

Everything else I totally agree on. I updated the draft with the code that makes the test pass, but this is obviously missing the check if the secret lives in a different account. Not sure if such a check is possible here.

Reply:
The problem we're currently dealing with is that we want to write an `if` for the case when the value passed is a Token, similar to what we have [here](https://github.com/aws/aws-cdk/blob/f3a830cb0d5b68e8f402791c3aaa5d1bcf2df673/packages/%40aws-cdk/aws-codebuild/lib/project.ts#L806-L806) for the case when the value is an ARN.

The problem is that there's nothing to compare the account of the Project to. Just adding the KMS permissions unconditionally here is a non-starter, IMO.

If we allowed passing a `Secret` as the value of the environment variable, then you can compare the account of the Project with the account of the passed Secret, and solve this problem this way.

Does this make sense?

Reply:
Makes sense 👍🏿

Reply:
I'll have a look into that approach tomorrow and will update this draft.

In my head the implementation will need another case for when the secret is neither a `Token` nor an `Arn`, but a `Secret`. Let me know if you disagree 😊

Reply:
Yep, makes sense.